### ISSUE TITLE
Filter stuck active when use camelCae name

### ISSUE THREAD

**Issue by kismonguz** [Laravel-Backpack/CRUD#2829]
# Bug report

### What I did

```
$this->crud->addFilter(['type' => 'text','name' => 'displayName','label' => 'Display name'],
false,
function($value){
    $this->crud->addClause('where', 'displayName', 'LIKE', "%$value%");
});
```
In the list screen first I typed anything into this filter, and after that I clicked the "Remove filters" link.

### What I expected to happen

The "Display name" filter shouldn't be active, and the filtered value should be empty.

### What happened

The filter still get active, and the filtered value does not appears.

### What I've already tried to fix it

I think I found what couse this case.
At the begining of the filter template (resources/views/crud/filters/text.blade.php:3) the "filter-name" attribute uses the raw name
```
<li filter-name="{{ $filter->name }}"
	filter-type="{{ $filter->type }}"
	class="nav-item dropdown {{ Request::get($filter->name) ? 'active' : '' }}">
```
Later at row 61 the event listener uses the slugged name. 
```
$('li[filter-name={{ Str::slug($filter->name) }}]').on('filter:clear', function(e) {
	$('li[filter-name={{ $filter->name }}]').removeClass('active');
	$('#text-filter-{{ Str::slug($filter->name) }}').val('');
});
```
The Laravels's Str::slug() destroys the camelCase notation and convert "displayName" to "displayname"

There is same issue with other filters. If I have right, that would be nice if the code uses same version of filter name in whole filter template.

Thanks!

### Backpack, Laravel, PHP, DB version

When I run ```php artisan backpack:version``` the output is:
### PHP VERSION:
PHP 7.2.30-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Apr 19 2020 07:50:50) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Xdebug v2.6.1, Copyright (c) 2002-2018, by Derick Rethans
    with Zend OPcache v7.2.30-1+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies

### LARAVEL VERSION:
v7.11.0@f4563bd2e0875c59a1f7967abdbe5cef7f240117

### BACKPACK VERSION:
4.1.4@4303aa07ec0c597abfcfe83896558ca6e98f0d6a

**Comment by tabacitu** [comment]
Thanks for reporting this @kismonguz ! We'll investigate and fix ASAP.

**Comment by kismonguz** [comment]
Your welcome!

I really like this project!

**Comment by tabacitu** [comment]
Just merged #2872 which fixes this - thanks again for raising the issue @kismonguz .

**Comment by tabacitu** [comment]
@kismonguz looks like fixing this issue has caused some unexpected problems. Can you please give #3065 a go, and let me know if after that change you're still having any issues? It'd really help us out.
